services/httppost/service.go

```diff
@@ -243,8 +249,17 @@ func (h *handler) Handle(event alert.Event) {
 		return
 	}
 
-	// Execute the request
+	// Set content type
 	req.Header.Set("Content-Type", "application/json")
+
+	// Set timeout
+	if h.timeout > 0 {
+		ctx, cancel := context.WithTimeout(req.Context(), h.timeout)
+		defer cancel()
+		req = req.WithContext(ctx)
+	}
```

Comment:
This code is repeated above as well. Since all code to get a request goes through https://github.com/sputnik13/kapacitor/blob/50f47625628a0937873ec60cda54c4b6d880ab0a/services/httppost/service.go#L50 why not move the logic to set a Timeout context on the request into that method?

The function will probably need  a new parameter for the desired timeout since the source of the timeout varies, but that seems reasonable.

Reply:
I'm not sure that's necessarily better than setting the timeout locally.  While there would be less code I think clarity will suffer since the function interface for NewHTTPRequest will have to return a cancel functor for the context, but it could be misconstrued as a cancel functor for the request.  I wish there was a means to set the timeout on the Request without having to pass in a context object.

If you disagree and think it's clear enough I can make the change.

Reply:
I see, lets leave it how it is then. That makes sense.